packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -18,21 +18,43 @@ import { assert, performance } from "@fluidframework/common-utils";
  */
 export const latencyThreshold = 5000;
 
+interface IOpPerfTelemetryProperties {
+    /** Track how long op is sitting in inbound queue until it is processed */
+    durationInboundQueue: number | undefined;
+    /** Measure time outbound op is sitting in queue due to active batch */
+    durationOutboundQueue: number | undefined;
+    /** Time spent between DeltaManager's inbound "push" event until the DeltaManager "op" event */
+    durationInboundToProcessing: number | undefined;
+    /** Length of the DeltaManager's inbound queue at the time of the processing */
+    lenghtInboundQueue: number | undefined;
+}
+/**
+ * We collect the timing at various moments in time during the op processing.
+ */
+enum OpTimings {
+    opSendTimeForLatencyStatistics,
+    opTimeSittingInboundQueue,
+    opTimeInboundPushEvent,
+}
 class OpPerfTelemetry {
     private pongCount: number = 0;
     private pingLatency: number | undefined;
 
     // Collab window tracking. This is timestamp of %1000 message.
     private sequenceNumberForMsnTracking: number | undefined;
     private msnTrackingTimestamp: number = 0;
-
-    // To track round trip time for every %1000 client message.
-    private opSendTimeForLatencyStatistics: number | undefined;
-
+    // To track round trip time for every %500 client message.
     private clientSequenceNumberForLatencyStatistics: number | undefined;
 
-    private opTimeSittingInInboundQueue: number | undefined;
-    private durationSittingInInboundQueue: number | undefined;
+    private readonly opProcessingTimes: number | undefined[] = [];
+
+    // Performance Data to be reported for ops round trips and processing.
+    private opPerfData: IOpPerfTelemetryProperties = {
+        durationInboundQueue: undefined,
```

Comment:
All the members are undefined by default, so you can remove them and just have = {}